src/workflow/ArcanistLandWorkflow.php

```diff
@@ -291,7 +313,8 @@ public function run() {
         ->setShouldKeep($this->keepBranch)
         ->setShouldSquash($this->useSquash)
         ->setShouldPreview($this->preview)
-        ->setBuildMessageCallback(array($this, 'buildEngineMessage'));
+        ->setBuildMessageCallback(array($this, 'buildEngineMessage'))
+        ->setRevision($this->getRevisionDict());
```

Comment:
Again, are you sure it will not break default behavior if revision is not set? Maybe it would be worth add check if submit queue is enabled:

```
if ($this->shouldUseSubmitQueue && !$this->tbr) {
  $engine->setRevision($this->getRevisionDict());
}
```